File: swift/abc021A.swift
var l: [Int] = []
var c = [8, 4, 2, 1]
let a = readLine()!
var n = Int(a)!
var i = 0
for a in c {
  if (n == 0) {
    break
  }
  if (a <= n){
    n = n - a
    l.append(a)
    i = i + 1
  }
}
print (i)
for b in l {
  print(b)
}

File: C#/abc021A.cs
using System;

class main{
  public static int Main(){
    int[] l = new int[4];
    int[] c = { 8, 4, 2, 1 };
    int i = 0;
    int j = 0;
    int n = int.Parse(Console.ReadLine());
    while(n != 0){
      if(c[j] <= n){
       n = n - c[j];
       l[i] = c[j];
       i = i + 1;
      }
      j = j + 1;
    }
    j = 0;
    Console.WriteLine(i);
    while(j < i){
      Console.WriteLine(l[j]);
      j = j + 1;
    }
    return 0;
  }
}
